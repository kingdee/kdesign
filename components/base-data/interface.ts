import { tuple } from '../_utils/type'

export const Modes = tuple('multiple', 'single')
export type Mode = typeof Modes[number]

export type SelectValue = string | string[] | number | number[] | IOptionsProps[] | null

export interface IAdvancedSelectorProps<T = SelectValue> {
  prefixCls?: string // 样式前缀
  className?: string // 样式名
  children?: React.ReactNode
  style?: React.CSSProperties // 内联样式
  disabled?: boolean // 禁用
  dropdownStyle?: React.CSSProperties // 下拉样式
  notFoundContent?: React.ReactNode // 列表为空时文本
  placeholder?: string // 选择框默认文字
  showFrequent?: boolean // 是否显示历史记录与我的收藏面板
  showCollectIcon?: boolean // 是否显示收藏按钮
  showDetailIcon?: boolean // 是否显示详情按钮
  suffixIcon?: React.ReactNode // 选择框后缀图标
  mode?: Mode // 模式
  value?: IOptionsProps[] | IOptionsProps // 指定选中的条目
  columns?: [] // 下拉列表的展示列
  options?: [] // 下拉列表的数据
  size?: number // 下拉列表显示数量
  optionLabelProp?: string // 回填到选择框的属性
  dropdownFooterRender?: React.ReactNode // 自定义下拉框footer内容
  visible?: boolean
  collectList?: [] // 收藏列表
  historyList?: [] // 我的历史
  loading?: boolean // 加载中状态
  searchField?: string // 搜索字段 默认 all（全部字段）
  onChange?: () => void // 选中option调用此函数
  onSearch?: (val: string) => void // input的value变化时调用
  onSelect?: (value: T extends (infer I)[] ? I : T, option: React.ReactElement<any>) => void
  onShowDetail?: (value: IOptionsProps[]) => void
  onShowMore?: () => void
  onCollect?: (flag: boolean, value: IOptionsProps) => void
}

export interface IIndxPosListProps {
  end: number
  start: number
}

export interface IColumnsProps {
  title?: string
  key: string
}

export interface IOptionsProps {
  value?: string | number
  label?: string
}

export interface ISearchInfoProps {
  searchValue?: string
  previousInputValue: string
  previousEditValue?: string
  searchIndex?: number
  selectionStart?: number
  deleteEndIndx: number
  editOptions?: any[]
}
