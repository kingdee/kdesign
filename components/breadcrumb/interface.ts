import { ReactNode } from 'react'
import { DropDownProps } from '../dropdown'
import { tuple } from '../_utils/type'

export const ColorModelTypes = tuple('emphasize', 'weaken')
export type ColorModelType = typeof ColorModelTypes[number]

export interface IBreadcrumbProps {
  style?: Record<string, unknown> // 内联样式
  className?: string // 类名
  colorModel?: ColorModelType // 颜色模式，默认是 'emphasize'
  prefixCls?: string // 样式前缀
  items: Array<IBreadcrumbItem> // 面包屑导航的各项，支持传入 ReactNode 类型
  separator?: ReactNode // 分隔符，默认是 '/'
  children?: ReactNode // 子元素
  onItemClick?: (item: IBreadcrumbItem, index: number) => void // 点击面包屑导航的回调函数
}

export interface IBreadcrumbItems {
  item: IBreadcrumbItem
  index: number
  separator?: ReactNode
  openEllipsis?: boolean
  onItemClick?: (item: IBreadcrumbItem, index: number) => void // 点击面包屑导航的回调函数
}

export interface IBreadcrumbItem {
  className?: string // 类名
  title?: ReactNode
  dropdownProps?: DropDownProps
  href?: string // 链接地址，不能和path共用
  path?: string // 拼接地址，不能和href共用
  icon?: ReactNode
}

export interface IItemsWidth {
  width: number
  index: number
}
