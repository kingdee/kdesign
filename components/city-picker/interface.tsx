import * as React from 'react'
import { tuple } from '../_utils/type'

export type City = {
  id?: string | number
  country?: string
  province?: string
  name?: string
  [key: string]: any
}

export const BorderTypes = tuple('none', 'underline', 'bordered')
export type BorderType = typeof BorderTypes[number]

export const Sizes = tuple('large', 'middle', 'small') // 选择框大小
export type Size = typeof Sizes[number]

export type Type = 'domestic' | 'foreign'

export type CityList = Array<City>

export interface CityPickerProps {
  prefixCls?: string // 样式前缀
  size?: Size // 尺寸
  borderType?: BorderType // 边框类型
  notFoundContent?: React.ReactNode // 列表为空时文本
  showArrow?: boolean // 显示下拉小箭头
  defaultOpen?: boolean // 默认是否展开下拉列表
  allowClear?: boolean // 允许清除内容
  placeholder?: React.ReactNode // 选择框默认文字
  type?: Type
  loading?: boolean
  domesticList?: CityList
  foreignList?: CityList
  commonList?: CityList
  className?: string
  disabled?: boolean
  style?: React.CSSProperties
  title?: string
  clearIcon?: React.ReactNode // 清除图标
  suffixIcon?: React.ReactNode // 选择框后缀图标
  optionHighlightProps?: string // 搜索时 根据该属性值高亮城市名
  value?: string | number
  defaultValue?: string | number
  description?: React.ReactNode
  showDescription?: boolean
  onChange?: (value: string | number, city: City) => void
  onBlur?: (e: React.ChangeEvent<HTMLSpanElement>) => void
  onFocus?: () => void
  onSearch?: (value: string) => void
  onClear?: () => void
  itemRender?: (city: City) => React.ReactNode
  onTabPaneChange?: (id: Type) => void
}

export interface ICityPickerOptionProps {
  disabled?: boolean
  city?: City
  value?: string | number
  children?: React.ReactNode
  className?: string
  style?: React.CSSProperties
  onChangeSelect?: (city: City | null) => void
  renderCityInfo?: (city: City, flag?: boolean, symbol?: string) => React.ReactNode
  itemRender?: (city: City) => React.ReactNode
}
