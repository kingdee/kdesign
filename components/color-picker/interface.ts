import { tuple } from '../_utils/type'
import { PopperProps } from '../_utils/usePopper'

export const ColorTypes = tuple('HEX', 'HSB', 'RGB', 'HSL', 'themeColor')
export type ColorType = typeof ColorTypes[number]
export interface IColorTypesObj {
  type: ColorType
  value: string
}

export const BorderTypes = tuple('underline', 'bordered')
export type BorderType = typeof BorderTypes[number]
export interface IColorPickerProps extends PopperProps {
  // 设置颜色选择器的值，可以是十六进制颜色值，RGB颜色值，HSL颜色值，或者颜色名称
  value: string
  // 设置颜色选择器的默认值，可以是十六进制颜色值，RGB颜色值，HSL颜色值，或者颜色名称
  defaultValue: string
  // 选择器输入框类名
  className: string
  // 选择输入框样式
  style: React.CSSProperties
  // 输入框类型
  borderType: BorderType
  // 输入框右侧的图标
  suffixIcon: (rgbColor: string) => React.ReactNode
  // 功能色名称
  functionalColorName: string
  // 功能色色值
  functionalColor: string
  // 功能色开关的名称
  switchName: { name: string; internationalName?: string }
  // 是否跟随功能色,定义了功能色才能控制显示跟随主题色开关
  showSwitch: boolean
  // 是否展示色彩格式转换框
  showColorTransfer: boolean
  // 是否展示系统预设颜色
  showPresetColor: boolean
  // 占位字符
  placeholder: string
  // 用户自定义的预设颜色
  presetColor: string[]
  // 是否展示拾色容器
  showColorPickerBox: { showBox: boolean; showHue: boolean; showOpacity: boolean }
  // 颜色选择面板显隐配置
  defaultOpen: boolean
  // 当颜色选择器的值发生变化时触发的回调函数
  onChange: (inputValue: string, formatColorArr: Array<IColorTypesObj>) => void
}

export interface IColorPickerPanelProps {
  alpha: number
  alphaNoVerifyVal: string
  clickedColorIndex?: number
  colTypeArr: Array<IColorTypesObj>
  correctColorValue: string
  currentColorType: IColorTypesObj['type']
  functionalColor?: string
  functionalColorName: string
  switchName: { name: string; internationalName?: string }
  inputRef: any
  isFollow: boolean
  presetColor?: string[]
  showSwitch?: boolean
  showColorTransfer?: boolean
  showPresetColor?: boolean
  showColorPickerBox?: { showBox?: boolean; showHue?: boolean; showOpacity?: boolean }
  showColorPickerPanel: boolean
  value: string
  setCurrentColorType: (currentColorType: IColorTypesObj['type']) => void
  setColTypeArr: (colTypeArr: Array<IColorTypesObj>) => void
  setClickedColorIndex: (clickedColorIndex: number) => void
  setIsFollow: (isFollow: boolean) => void
  setAlphaNoVerifyVal: (NoVerifyVal: string) => void
  setAlpha: (alpha: number) => void
  setCorrectColorValue: (colorValue: string) => void
  setInputColorValue: (colorValue: string) => void
  setShowPanel: (showPanel: boolean) => void
  onChange?: (inputValue: string, formatColorArr: Array<IColorTypesObj>) => void
}
