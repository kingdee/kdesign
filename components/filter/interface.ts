import React from 'react'
import { Menu } from '../dropdown/menu'
import { IQuickSearchProps } from '../search'
import { DirectionType } from '../config-provider/ConfigContext'

type TabKey = 'condition' | 'scheme'

export type TFilterValue = {
  [key: string]: string[] | IScheme
}

export type TOption = {
  value?: string
  label?: string
  component?: React.ReactNode
}

export interface ICondition {
  key: string
  label: string
  options: TOption[]
  required?: boolean
  multiple?: boolean
  addition?: React.ReactNode
}

export interface IField {
  name: string
  label: string
  required?: boolean
  operators: React.ReactElement
  assignOptions: Record<string, React.ReactElement>
}

export interface ISchemeItem {
  fieldName: string
  value: string
  operator: string
  required?: boolean
}

export interface IScheme {
  name?: string
  preset?: boolean
  items: ISchemeItem[]
}

export interface IFilterProps {
  spread?: boolean
  fields?: IField[]
  className?: string
  schemes?: IScheme[]
  value?: TFilterValue
  defaultTabKey?: TabKey
  defaultSpread?: boolean
  title?: React.ReactNode
  conditions?: ICondition[]
  search?: IQuickSearchProps
  style?: React.CSSProperties
  defaultValue?: TFilterValue
  onSchemeSave?: (scheme: IScheme) => void
  onSpreadChange?: (spread: boolean) => void
  getSchemeSettingMenu?: (scheme: IScheme) => typeof Menu
  onChange?: (value: TFilterValue, condition?: ICondition, option?: TOption) => void
}

export interface IConditionProps {
  tab: string
  prefixCls: string
  value: TFilterValue
  conditions: ICondition[]
  FilterLangMsg: Record<string, any>
  onConditionRemove: (key: string, e: React.MouseEvent) => void
  onConditionChange: (value: TFilterValue, condition?: ICondition, option?: TOption) => void
}

export interface ISchemeProps {
  tab: string
  fields: IField[]
  prefixCls: string
  schemes: IScheme[]
  direction: DirectionType
  value: { scheme: IScheme }
  FilterLangMsg: Record<string, any>
  onSchemeSave: (scheme: IScheme) => void
  onSchemeSearch: (scheme: IScheme) => void
  getSchemeSettingMenu?: (scheme: IScheme) => void
}
