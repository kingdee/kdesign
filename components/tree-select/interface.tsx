import React, { ReactNode } from 'react'
import { tuple } from '../_utils/type'
import { PopperProps } from '../_utils/usePopper'
import { TreeNodeData } from '../tree'
export const SelectSizes = tuple('large', 'middle', 'small') // 选择框大小
export type SelectSize = typeof SelectSizes[number]

export const BorderTypes = tuple('none', 'underline', 'bordered')
export type BorderType = typeof BorderTypes[number]

export const Modes = tuple('multiple', 'single')
export type Mode = typeof Modes[number]
export interface AbstractSelectProps extends PopperProps {
  prefixCls?: string // 样式前缀
  className?: string // 样式名
  size?: SelectSize // 尺寸
  borderType?: BorderType // 边框类型
  notFoundContent?: React.ReactNode | null // 列表为空时文本
  showSearch?: boolean // 可搜索
  showArrow?: boolean // 显示下拉小箭头
  defaultOpen?: boolean // 默认是否展开下拉列表
  allowClear?: boolean // 允许清除内容
  disabled?: boolean // 禁用
  style?: React.CSSProperties // 内联样式
  placeholder?: string | React.ReactNode // 选择框默认文字
  dropdownClassName?: string // 下拉样式名
  dropdownStyle?: React.CSSProperties // 下拉样式
  getPopupContainer?: (triggerNode: HTMLElement) => HTMLElement // 菜单渲染父节点
  dropdownRender?: (menu: React.ReactElement) => React.ReactElement // 自定义下拉框渲染内容
  id?: string
  clearIcon?: React.ReactNode // 清除图标
  searchIcon?: React.ReactNode // 搜索图标
  suffixIcon?: React.ReactNode // 选择框后缀图标
  maxTagCount?: number // 最多显示tag数
  labelInValue?: boolean // 每个选项的 label 包装到 value 中
  maxTagPlaceholder?: React.ReactNode | ((omittedValues: TreeNodeData[]) => React.ReactNode) // 隐藏 tag 时显示的内容
}

export type ITreeProps = {
  treeData?: TreeNodeData[]
  treeDefaultExpandAll?: boolean
  treeDefaultExpandedKeys?: string[]
  treeExpandedKeys?: string[]
  treeIcon?: ReactNode | ((props: any) => ReactNode)
  virtual?: boolean
  checkStrictly?: boolean
  filterTreeNode?: (treeNode: TreeNodeData, searchValue: string) => boolean
  treeNodeFilterProp?: string
  treeNodeLabelProp?: string
  expandOnClickNode?: boolean
  treeCheckStrictly?: boolean
  treeExpandOnClickNode?: boolean
  showTreeIcon?: boolean
  treeLoadData?: () => void
}

export type TreeSelectValue = string | string[] | number | number[] | undefined
export interface ITreeSelectProps<T extends TreeSelectValue> extends AbstractSelectProps, ITreeProps {
  value?: T
  defaultValue?: T
  mode?: Mode
  autoFocus?: boolean
  onChange?: (value: T, treeNode: TreeNodeData) => void
  onSelect?: (value: T extends (infer I)[] ? I : T, { checked, node, event }: any) => void
  onBlur?: (value?: T) => void
  onFocus?: () => void
  onSearch?: (value?: string) => void
  onClear?: () => void
  tagRender?: (props: TagProps) => void
}

export interface TagProps {
  value?: string
  label?: React.ReactNode
  disabled: boolean
  size: SelectSize
  onClose: () => void
}
