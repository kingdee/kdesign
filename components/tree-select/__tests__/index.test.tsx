import React from 'react'
import { render, mount } from 'enzyme'
import ConfigProvider from '../../config-provider/index'
import TreeSelect from '../index'
import { BorderTypes, SelectSizes, Modes } from '../interface'
import mountTest from '../../../tests/shared/mountTest'

const treeData = [
  {
    key: '0',
    title: '0',
    children: [
      {
        key: '0-0',
        title: '0-0',
        children: [],
      },
      {
        key: '0-1',
        title: '0-1',
        children: [],
      },
    ],
  },
  {
    key: '1',
    title: '1',
    children: [
      {
        key: '1-0',
        title: '1-0',
        children: [],
      },
      {
        key: '1-1',
        title: '1-1',
        children: [
          {
            key: '1-1-0',
            title: '1-1-0',
            children: [
              {
                key: '1-1-0-0',
                title: '1-1-0-0',
                children: [],
              },
            ],
          },
          {
            key: '1-1-1',
            title: '1-1-1',
            children: [
              {
                key: '1-1-1-0',
                title: '1-1-1-0',
                children: [],
              },
            ],
          },
        ],
      },
      {
        key: '1-2',
        title: '1-2',
        children: [],
      },
    ],
  },
  {
    key: '2',
    title: '2',
    children: [],
  },
  {
    key: '3',
    title: '3',
    children: [],
  },
]

const defaultTreeSelectProps = {
  treeData: treeData,
  defaultOpen: true,
  onFocus: jest.fn(),
  onBlur: jest.fn(),
  onChange: jest.fn(),
  onClear: jest.fn(),
  onSearch: jest.fn(),
  onSelect: jest.fn(),
  onTreeExpand: jest.fn(),
}

describe('TreeSelect', () => {
  // 1. mount test
  describe('1. mount test', () => {
    mountTest(() => <TreeSelect />)
  })

  // 2. render test
  describe('2. render test', () => {
    it('renders correctly', () => {
      expect(render(<TreeSelect />)).toMatchSnapshot()

      BorderTypes.forEach((type) => {
        expect(render(<TreeSelect borderType={type} />)).toMatchSnapshot()
      })

      SelectSizes.forEach((type) => {
        expect(render(<TreeSelect size={type} />)).toMatchSnapshot()
      })

      Modes.forEach((type) => {
        expect(render(<TreeSelect mode={type} />)).toMatchSnapshot()
      })
    })
  })

  // 5. displayName
  describe('5. displayName', () => {
    it('should have displayName static property', () => {
      const TestTreeSelect = mount(<TreeSelect />)
      expect((TestTreeSelect.type() as any).displayName).toBe('TreeSelect')
    })
  })

  // 6. class state
  describe('6. class state', () => {
    it('should borderType class use right', () => {
      BorderTypes.forEach((type) => {
        const TestTreeSelect = mount(<TreeSelect borderType={type} />)
        expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName(
          `.kd-tree-select-${type === 'none' ? 'borderless' : type}`,
        )
      })
    })

    it('should size class use right', () => {
      SelectSizes.forEach((type) => {
        const TestTreeSelect = mount(<TreeSelect size={type} />)
        expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName(`.kd-tree-select-size-${type}`)
      })
    })

    it('should mode and disabled class use right', () => {
      Modes.forEach((type) => {
        const TestTreeSelect = mount(<TreeSelect mode={type} disabled />)
        expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName(`.kd-tree-select-${type}`)
        expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName(`.kd-tree-select-${type}-disabled`)
      })
    })

    it('should className style use right', () => {
      const TestTreeSelect = mount(<TreeSelect className="kd-test" style={{ color: 'red' }} />)
      expect(TestTreeSelect.find('.kd-tree-select')).toHaveClassName('.kd-test')
      expect(TestTreeSelect.find('.kd-tree-select')).toHaveStyle('color', 'red')
    })

    it('should dropdownClassName use right', () => {
      const TestTreeSelect = mount(<TreeSelect dropdownClassName="kd-dropdown-test" defaultOpen />)
      expect(TestTreeSelect.find('.kd-tree-select-dropdown')).toHaveClassName('.kd-dropdown-test')
    })
  })

  // 7.component interaction(event)
  describe('7.component interaction(event)', () => {
    it('test function called successfully', () => {
      let val = ''
      const onChange = jest.fn((key) => {
        val = key
      })
      const TestTreeSelect = mount(
        <TreeSelect {...defaultTreeSelectProps} onChange={onChange} allowClear treeData={treeData} />,
      )
      TestTreeSelect.find('input').simulate('focus')
      expect(defaultTreeSelectProps.onFocus).toHaveBeenCalled()

      TestTreeSelect.find('input').simulate('blur')
      expect(defaultTreeSelectProps.onBlur).toHaveBeenCalled()

      TestTreeSelect.find('input').simulate('change', { target: { value: '12' } })
      expect(defaultTreeSelectProps.onSearch).toHaveBeenCalled()

      TestTreeSelect.find('.kd-tree-node-item-0').find('.kd-tree-node-icon').simulate('click')
      expect(defaultTreeSelectProps.onTreeExpand).toHaveBeenCalled()

      TestTreeSelect.find('.kd-tree-node-item-0-0').at(0).simulate('click')
      expect(defaultTreeSelectProps.onSelect).toHaveBeenCalled()
      expect(val).toBe('0-0')

      TestTreeSelect.find('.kd-tree-select-icon-clear').simulate('click')
      expect(defaultTreeSelectProps.onClear).toHaveBeenCalled()
    })
  })

  // 8.config provider
  describe('8.config provider', () => {
    it('should config use config provider', () => {
      const treeSelectConfig = {
        compDefaultProps: {
          TreeSelect: {
            size: 'small',
          },
        },
      }
      const TestTreeSelect = mount(
        <ConfigProvider value={treeSelectConfig}>
          <TreeSelect />
        </ConfigProvider>,
      )
      expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName('.kd-tree-select-size-small')
    })
  })

  // 9. ref test
  describe('9. ref test', () => {
    it('should get TreeSelect element from ref', () => {
      const ref = React.createRef()
      mount(<TreeSelect ref={ref}></TreeSelect>)
      expect(ref.current instanceof HTMLElement).toBe(true)
    })
  })

  // 10. api test
  describe('10. api test', () => {
    it('autoFocus', () => {
      // eslint-disable-next-line jsx-a11y/no-autofocus
      const TestTreeSelect = mount(<TreeSelect autoFocus={true} />)
      expect(TestTreeSelect.find(`.kd-tree-select-wrapper`)).toHaveClassName('.kd-tree-select-single-focused')
    })

    it('display the correct react node', () => {
      const TestTreeSelect = mount(
        <TreeSelect
          {...defaultTreeSelectProps}
          treeData={treeData}
          allowClear
          clearIcon="clearIcon-test"
          value={'0'}
          placeholder="placeholder-test"
          dropdownRender={() => <span>dropdownRender-test</span>}
        />,
      )

      // clearIcon
      expect(TestTreeSelect.find('.kd-tree-select-icon-clear').text()).toBe('clearIcon-test')
      // dropdownRender
      expect(TestTreeSelect.find('.kd-tree-select-dropdown span').text()).toBe('dropdownRender-test')

      // suffixIcon
      TestTreeSelect.setProps({ suffixIcon: 'suffixIcon-test' })
      expect(TestTreeSelect.find('.kd-tree-select-icon-arrow')).toHaveText('suffixIcon-test')

      // showArrow
      expect(TestTreeSelect.find('.kd-tree-select-icon-arrow').length).toBe(1)
      TestTreeSelect.setProps({ showArrow: false })
      expect(TestTreeSelect.find('.kd-tree-select-icon-arrow').length).toBe(0)

      // showSearch
      expect(TestTreeSelect.find('input').props().readOnly).toBe(false)
      TestTreeSelect.setProps({ showSearch: false })
      expect(TestTreeSelect.find('input').props().readOnly).toBe(true)

      TestTreeSelect.setProps({ mode: 'multiple' })
      expect(TestTreeSelect.find('.kd-tree-select-selection-tag').length).toBe(1)

      // tagRender
      TestTreeSelect.setProps({ tagRender: () => <span className="tagRender-test"></span> })
      expect(TestTreeSelect.find('.tagRender-test').length).toBe(1)

      TestTreeSelect.setProps({
        value: ['2', '3'],
        maxTagCount: 1,
        maxTagPlaceholder: <span className="kd-maxTagPlaceholder">kd</span>,
      })
      TestTreeSelect.update()
      // maxTagCount
      expect(TestTreeSelect.find('.kd-tree-select-selection-tag').length).toBe(1)
      // maxTagPlaceholder
      expect(TestTreeSelect.find('.kd-maxTagPlaceholder').length).toBe(1)

      // notFoundContent
      TestTreeSelect.setProps({ treeData: [], notFoundContent: 'notFoundContent-test', dropdownRender: undefined })
      expect(TestTreeSelect.find('.kd-tree-select-dropdown').text()).toBe('notFoundContent-test')
    })

    it('getPopupContainer', () => {
      const wrapperRef = React.createRef() as any
      const popupContainer = mount(
        <div ref={wrapperRef}>
          <TreeSelect defaultOpen getPopupContainer={() => wrapperRef.current} />
        </div>,
      )
      expect(popupContainer.childAt(0).children().at(1).find('.kd-tree-select-dropdown').length).toBe(1)
    })

    it('should pass tree props to Tree component', () => {
      const TestTreeSelect = mount(
        <TreeSelect
          {...defaultTreeSelectProps}
          treeIcon="treeIcon"
          treeData={treeData}
          treeDefaultExpandAll
          treeDefaultExpandedKeys={[]}
          treeExpandedKeys={[]}
          treeCheckStrictly
          treeExpandOnClickNode
          treeLoadData={() => undefined}
          showTreeIcon
          filterTreeNode={() => true}
          virtual
        />,
      )
      const treeProps = [
        'treeData',
        'icon',
        'virtual',
        'defaultExpandAll',
        'defaultExpandedKeys',
        'checkStrictly',
        'expandedKeys',
        'icon',
        'showIcon',
        'switcherIcon',
      ]
      treeProps.forEach((prop) => {
        expect(TestTreeSelect.find('Tree').prop(prop)).toBeDefined()
      })
    })
  })
})
