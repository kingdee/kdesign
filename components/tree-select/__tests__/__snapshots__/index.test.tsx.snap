// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect 2. render test renders correctly 1`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 2`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-borderless kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 3`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 4`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-bordered kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 5`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-large kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 6`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 7`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-small kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 8`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-multiple kd-tree-select-placeholder"
    >
      <span
        class="kd-tree-select-selection-search"
        style="width:4px"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
        <span
          class="kd-tree-select-selection-search-mirror"
        >
           
        </span>
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;

exports[`TreeSelect 2. render test renders correctly 9`] = `
<div
  class="kd-tree-select"
>
  <span
    class="kd-tree-select-selector"
    tabindex="0"
  >
    <div
      class="kd-tree-select-underline kd-tree-select-size-middle kd-tree-select-wrapper kd-tree-select-single"
      title=""
    >
      <span
        class="kd-tree-select-selection-search"
      >
        <input
          class="kd-tree-select-selection-search-input"
          value=""
        />
      </span>
      <span
        class="kd-tree-select-placeholder"
      >
        请输入需要搜索的内容
      </span>
      <span
        class="kd-tree-select-suffix"
      >
        <span
          class="kd-tree-select-icon-arrow kd-tree-select-icon-arrow-down"
        >
          <i
            class="kdicon kdicon-arrow-down"
          />
        </span>
      </span>
    </div>
  </span>
</div>
`;
